### DISCUSSION TITLE
Can Next.js 13 score 100 on Google PageSpeed Insights?

### DISCUSSION THREAD

**Discussion by zwl1619** [vercel/next.js#47075] [Help]
### Summary

[Qwik](https://github.com/BuilderIO/qwik) framework can score 100 on Google PageSpeed Insights. There is an article discussing this topic：
https://dev.to/builderio/how-to-score-100-on-google-pagespeed-insights-on-mobile-2e0i

But this article was posted in 2021. My question is, today, can Next.js 13 score 100 on Google PageSpeed Insights?

**Comment by andrepichardo** [comment]
Hi!

Getting a score that is higher than 90 in Google’s PageSpeed is a broad topic with much ground to cover.

In Next.js 13 it is indeed possible to get such a high score, but there are some points to take into account to achieve that with this framework, such as:

1. SSR (server-side rendering)

2. SSG (static site generation)

3. The `<Image/>` component

4. Font Optimization

5. Dynamic Import

6. Script Optimization

7. Next.js Analytics

As long as you correctly use all these features from Next.js, you should expect to get a really good score from Google PageSpeed Insights! :)

Good luck and hope this answer helps.

**Comment by icyJoseph** [comment]
Sure you can, but unfortunately, most apps do not exist in a vacuum... 

In some pages, pagespeed insights might flag `gtag`... just google, `gtag vs pagespeed` or the like...

It has happen to me that an otherwise well performant app, gets sinked down by some 3rd party script, that loads an image just as big as the entire page.

I think @andrepichardo's is a good break down.

I'd add, to be careful with imports. Like:

```js
import crypto from "crypto";

export const specieEP = "https://pokeapi.co/api/v2/pokemon-species";
export const pokeEp = "https://pokeapi.co/api/v2/pokemon";

export const catchPokemon = async (id: number | string) => {
  // some stuff
  const uuid = crypto.randomUUID();

  // rest of stuff
```

And then on a custom hook, I had imported `specieEP`, that made the WebPack bundler bring about the entire `crypto-browserify`, which is about 400kB.

Moving the endpoints to their own file, fixed the issue. It is often the case that through imports, we bring more than we want, and it is not always easy to see.
